cookbook/upgrade/bundles.rst

```diff
@@ -0,0 +1,155 @@
+.. index::
+    single: Upgrading; Bundle; Major Version
+
+Upgrading a Third-Party Bundle for a Major Symfony Version
+==========================================================
+
+Symfony 3 was released on November 2015. Although this version doesn't contain
+any new feature, it removes all the backwards compatibility layers included in
+the previous 2.8 version. If your bundle uses any deprecated feature and it's
+published as a third-party bundle, applications upgrading to Symfony 3 will no
+longer be able to use it.
+
+Allow to Install Symfony 3 Components
+-------------------------------------
+
+Most third-party bundles define their Symfony dependencies using the ``~2.N`` or
+``^2.N`` constraints in the ``composer.json`` file. For example:
+
+.. code-block:: json
+
+    {
+        "require": {
+            "symfony/framework-bundle": "~2.3",
+            "symfony/finder": "~2.3",
+            "symfony/validator": "~2.3"
+        }
+    }
+
+These constraints prevent the bundle from using Symfony 3 components, so it makes
+it impossible to install it in a Symfony 3 based application. This issue is very
+easy to solve thanks to the flexibility of Composer dependencies constraints.
+Just replace ``~2.N`` by ``~2.N|~3.0`` (or ``^2.N`` by ``^2.N|~3.0``).
```

Comment:
Another commonly used version constraint is ``>=2.3`` (pretty bad, but I've seen quite a few bundles using it). Maybe we have to add a note/tip/caution to avoid using constraints like that and explicitely support only 2 and 3.